wpa_supplicant/ctrl_iface_zephyr.c

```diff
@@ -277,18 +277,37 @@ static void wpa_supplicant_global_ctrl_iface_receive(int sock, void *eloop_ctx,
 	size_t reply_len = 0;
 
 	buf = os_zalloc(CTRL_IFACE_MAX_LEN + 1);
-	if (!buf)
+	if (!buf) {
+		/* Do a dummy read to drain the data from the socket */
+		static unsigned char dummy[512];
```

Comment:
As the temp buffer is just meant for draining the socket, we can use much smaller buffer here (it will just loop more during read), and then we do not need to make this static (saves some memory). I suggest something like 32 or 64 bytes long buffer here.

Reply:
Sure, we can do that. But next set of changes is to move from heap->stack, hence made it draft, that makes all these hacks moot. But would require changing a lot of open-source code, so, still thinking on it.